runtime/logging.go

```diff
@@ -73,7 +74,22 @@ func (h *LoggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				bs, r.Body, err = readBody(r.Body)
 			}
 			if err == nil {
-				fields["req_body"] = string(bs)
+				if gzipReceived(r.Header) {
+					// the request is compressed
+					reader := bytes.NewReader(bs)
+					gzReader, err := gzip.NewReader(reader)
+					if err != nil {
+						h.logger.Error("Failed to read the compressed payload: %v", err.Error())
```

Comment:
h.logger.Error will let the control flow proceed, won't it? So we'd be passing a nil gzReader to `io.ReadAll`.

Reply:
I've rushed into the fix, you are right.
I've rewrote the way errors are handled.